### ISSUE TITLE
[REVIEW]: ReproZip

### ISSUE THREAD

**Issue by whedon** [openjournals/joss-reviews#107]
**Submitting author:** @remram44 (<a href="http://orcid.org/0000-0002-0524-2282">Rémi Rampin</a>)
**Repository:** <a href="https://github.com/ViDA-NYU/reprozip" target ="_blank">https://github.com/ViDA-NYU/reprozip</a>
**Version:** 1.0.8
**Editor:** @acabunoc
**Reviewer:** @stain
**Archive:** <a href="https://dx.doi.org/10.5281/zenodo.159604" target="_blank">https://dx.doi.org/10.5281/zenodo.159604</a>

## Status

[![status](http://joss.theoj.org/papers/b578b171263c73f64dfb9d040ca80fe0/status.svg)](http://joss.theoj.org/papers/b578b171263c73f64dfb9d040ca80fe0)

Status badge code:

```
HTML: <a href="http://joss.theoj.org/papers/b578b171263c73f64dfb9d040ca80fe0"><img src="http://joss.theoj.org/papers/b578b171263c73f64dfb9d040ca80fe0/status.svg"></a>
Markdown: [![status](http://joss.theoj.org/papers/b578b171263c73f64dfb9d040ca80fe0/status.svg)](http://joss.theoj.org/papers/b578b171263c73f64dfb9d040ca80fe0)
```
**Reviewers and authors**:

Please avoid lengthy details of difficulties in the review thread. Instead, please create a new issue in the <a href="https://github.com/ViDA-NYU/reprozip" target="_blank">target repository</a> and link to those issues (especially acceptance-blockers) in the review thread below. (For completists: if the target issue tracker is also on GitHub, linking the review thread in the issue or vice versa will create corresponding breadcrumb trails in the link target.)

### Conflict of interest

- [x] As the reviewer I confirm that there are no conflicts of interest for me to review this work (such as being a major contributor to the software).

### General checks

- [x] **Repository:** Is the source code for this software available at the <a target="_blank" href="https://github.com/ViDA-NYU/reprozip">repository url</a>?
- [x] **License:** Does the repository contain a plain-text LICENSE file with the contents of an [OSI approved](https://opensource.org/licenses/alphabetical) software license?
- [x] **Version:** Does the release version given match the GitHub release (1.0.8)?
- [x] **Authorship:** Has the submitting author (@remram44) made major contributions to the software?

### Functionality

- [x] **Installation:** Does installation proceed as outlined in the documentation?
- [x] **Functionality:** Have the functional claims of the software been confirmed?
- [x] **Performance:** Have any performance claims of the software been confirmed?

### Documentation

- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **Installation instructions:** Is there a clearly-stated list of dependencies? Ideally these should be handled with an automated package management solution.
- [x] **Example usage:** Do the authors include examples of how to use the software (ideally to solve real-world analysis problems).
- [x] **Functionality documentation:** Is the core functionality of the software documented to a satisfactory level (e.g. API method documentation)?
- [x] **Automated tests:** Are there automated tests or manual steps described so that the function of the software can be verified?
- [x] **Community guidelines:** Are there clear guidelines for third parties wishing to 1) Contribute to the software 2) Report issues or problems with the software 3) Seek support

### Software paper

- [x] **Authors:** Does the `paper.md` file include a list of authors with their affiliations?
- [x] **A statement of need:** Do the authors clearly state what problems the software is designed to solve and who the target audience is?
- [x] **References:** Do all archival references that should have a DOI list one (e.g. papers, datasets, software)?

**Comment by whedon** [comment]
Hello human, I'm @whedon. I'm here to help you with some common editorial tasks for JOSS.

For a list of things I can do to help you, just type:

```
@whedon commands
```

**Comment by stain** [comment]
Thanks for the submission,  @remram44 - [paper](https://github.com/ViDA-NYU/reprozip/blob/joss-paper/paper/paper.md) looks good!

I just need to complete the testing of the software - when I tried this weekend to rerun a reprozip with `chroot` I lost my terminal files in my actual `/dev` (!) - so need to try again and raise issue upstream if appropriate.

Using `pip install` was enough to get started, but I also want to just verify running from the [archived source](https://doi.org/10.5281/zenodo.159604).

**Comment by remram44** [comment]
Removing the folder you unpacked in will indeed remove files from `/dev`, since they are bind-mounted in there. We provide [a `destroy` command](http://reprozip.readthedocs.io/en/1.0.x/unpacking.html#removing-the-experiment-directory) that will deal with it, unmounting in the case of chroot, or destroying containers/VMs.

Thank you for your time and the quick review!

**Comment by stain** [comment]
@whedon commands

**Comment by whedon** [comment]
Here are some things you can ask me to do:

```
# List all of Whedon's capabilities
@whedon commands

# Assign a GitHub user as the reviewer of this submission
@whedon assign @username as reviewer

# List the GitHub usernames of the JOSS editors
@whedon list editors

# List of JOSS reviewers together with programming language preferences and domain expertise
@whedon list reviewers

# Change editorial assignment
@whedon assign @username as editor

# Set the software archive DOI at the top of the issue e.g.
@whedon set 10.0000/zenodo.00000 as archive

# Open the review issue
@whedon start review
```

:construction: **Important** :construction:

This is all quite new. Please make sure you check the top of the issue after running a @whedon command (you might also need to refresh the page to see the issue update).

**Comment by stain** [comment]
*Community guidelines*:

The documentation already have a [developer guide](https://reprozip.readthedocs.io/en/1.0.x/developerguide.html) and - deep enough down - links to mailing lists, but it is not made clear in the github repo itself how to contribute, so I've suggested to add that in ViDA-NYU/reprozip#225

I also propose to add a **License** section to the README file as well, as not everyone recognize the [BSD 3 Clause](https://opensource.org/licenses/BSD-3-Clause) when they see it just in the [LICENSE.txt](https://github.com/ViDA-NYU/reprozip/blob/master/LICENSE.txt)

**Comment by stain** [comment]
@remram44 Just the two above issues in community guidelines ViDA-NYU/reprozip#225 and to improve the references ViDA-NYU/reprozip#226 - otherwise I think it's good to go!

**Comment by stain** [comment]
Sorry, @remram44 not sure if I should tick the `accepted` label, I think @acabunoc as the editor will do that formally.

**Comment by acabunoc** [comment]
This looks great! Happy to accept this.

Thanks so much @stain & @remram44 🎉 🎈 🍰 🎁 

Flagging @arfon because I'm actually not sure what the last step is. In the meantime, could you make sure there's an archive of the reviewed software in Zenodo/figshare/other service and update this thread with the DOI of the archive?

**Comment by remram44** [comment]
We have a DOI through Zenodo: [10.5281/zenodo.159604](https://doi.org/10.5281/zenodo.159604).

**Comment by acabunoc** [comment]
Thanks for adding the link! Helpful.

**Comment by arfon** [comment]
@remram44 - I don't seem to be able to find the `paper.md` (and associated `paper.bib`) file anywhere in this repository? Do you need to merge the `joss-paper` branch?

**Comment by arfon** [comment]
> The paper in on the joss-paper branch, does it need to be on master? It can't be added to the DOI'd archive retroactively anyway.

I think it should _eventually_ be on master (if that's your main branch).

**Comment by remram44** [comment]
@arfon: Is further action required on our part? I see that we haven't been added to joss-papers yet.

**Comment by arfon** [comment]
> @arfon: Is further action required on our part? I see that we haven't been added to joss-papers yet.

Sorry, this slipped through. On reflection I would like to see the `joss-paper` branch merged before I finalize this submission. Does that sound OK @remram44?

**Comment by remram44** [comment]
I remember talking about this with @jakevdp but he didn't really give an answer. I think I like it better this way, in its own branch directly based off the reviewed version (1.0.8). There is no need to clutter the repository (CI scripts are already bad enough), and it will not be updated.

Your joss-papers repository is the authoritative one anyway, and I'll add a badge linking to it in my master branch's README. Is that acceptable?

**Comment by arfon** [comment]
OK no problem @remram44. As long as the `joss-papers` branch is permanent this is OK with me.

I've opened a PR on your repo (https://github.com/ViDA-NYU/reprozip/pull/227) with some changes that need to be made to the paper structure before we can finish processing this submission. Please let me know when you've made these changes 😁

**Comment by remram44** [comment]
I merged [your changes](https://github.com/ViDA-NYU/reprozip/pull/227) and @VickySteeves added the requested [paper.bib](https://github.com/ViDA-NYU/reprozip/blob/joss-paper/paper/paper.bib).

**Comment by arfon** [comment]
@stain many thanks for the review here and thanks for editing this @acabunoc.

@remram44 - your paper is now accepted into JOSS and your DOI is http://dx.doi.org/10.21105/joss.00107 🚀 🎉 💥
